### PR TITLE
InputText basic reusable component

### PR DESCRIPTION
**Pull request by javiercodixis**
**What**

- InputText basic reusable component.

**How**

- Using React Form.
- Passing props to the InputText component, such as label, if is multiline, etc.
- If inputs are empty, error messages are shown and form is not sent.

**Screenshot**
image
image

### CODE REVIEW

File: components/InputText.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+type InputProps = {
+  name: string;
+  label: string;
+  control: Control<any>;
+  multiline?: boolean;
+  rows?: number;
+  required?: boolean;
+};
+
+const InputText = ({ name, label, control, multiline, rows, required }: InputProps) => {
```
Comment (bakycodixis) [inline components/InputText.tsx]: Several comments:
* Prefer using a different naming convention: `TextInput`. If you add prefix `Input`, then everything in your input folder will be starting with `Input` (hard to quickly find the file you need).
* Prefer extending the `TextFieldProps` coming from `@mui/material/TextField`. This way you don't need to specify anything in your custom type that is already included in `TextFieldProps`.

File: components/InputText.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  return (
+    <Controller
+      name={name as any}
+      control={control}
+      rules={{ required: `${label} is required`, validate: value => value.trim() !== '' || `${label} should not be empty` }}
```
Comment (bakycodixis) [inline components/InputText.tsx]: As we discussed earlier, `rules` value should be dynamic here. I should be able to pass whatever rules I want into this component.

File: components/InputText.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            {...field}
+            label={label}
+            multiline={multiline || false}
+            rows={rows || 1}
+            variant="outlined"
+            fullWidth
+            required={required}
+            error={fieldState.invalid}
+          />
+          <FormHelperText error={fieldState.invalid}>{fieldState.error?.message}</FormHelperText>
```
Comment (bakycodixis) [inline components/InputText.tsx]: We don't want to always render `FormHelperText` as it has padding. It adds unnecessary empty space on the bottom if you have no errors.

File: app/page.tsx [hunk 4]
```diff
@@ -1,75 +1,96 @@
...
-      phone: '',
-      whiskey: 'Whiskey Option 1',
-      quantity: 1,
-      address: '',
-      ageVerified: false,
-    },
-  });
-
-  const whiskeyOptions = ['Whiskey Option 1', 'Whiskey Option 2', 'Whiskey Option 3'];
+export default function Page() {
```
Comment (bakycodixis) [inline app/page.tsx]: Do you have `ESLint` working? This type of component declaration should not be allowed.

**Review by bakycodixis — CHANGES_REQUESTED** [review]

File: app/page.tsx (hunk 4 above)
Comment (javiercodixis) [reply app/page.tsx]: image
About this I have the arrow function definition, right? but for some reason is not creating a warning

File: app/page.tsx [hunk 5]
```diff
@@ -1,75 +1,99 @@
...
-  quantity: number;
-  address: string;
-  ageVerified: boolean;
-};
+const whiskeyOptions = ['Whiskey Option 1', 'Whiskey Option 2', 'Whiskey Option 3'];
 
-export default function WhiskeyPurchaseForm() {
-  const { register, handleSubmit, control } = useForm<FormValues>({
+const Page = () => {
+  const { handleSubmit, control } = useForm<FormValues>({
```
Comment (bakycodixis) [inline app/page.tsx]: Note: we don't want to pass `defaultValues` to `useForm`. We let the form initialize by itself. If TypyScript is complaining about missing values, let me know.

File: components/TextInput.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { Controller, Control } from 'react-hook-form';
+import TextField, { TextFieldProps } from '@mui/material/TextField';
+import { FormValues } from '@/types';
+
+type TextInputProps = TextFieldProps &  {
+  name: keyof FormValues;
+  label: string;
```
Comment (bakycodixis) [inline components/TextInput.tsx]: 1. Do we need `label` and `fullWidth`? They should be included in `TextFieldProps` already. 
2. I would recommend you to manage input widths from the `page.tsx` file by wrapping inputs with `Grid`. If I ever want to use `1/3` or `1/4` width input, then by your example I would need to add additional props to each input type. Not flexible.

File: components/TextInput.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  control: Control<FormValues>;
+  rules?: object;
+  fullWidth?: boolean;
+  halfWidth?: boolean;
+}
+
+const TextInput: React.FC<TextInputProps> = ({ name, label, control, rules, fullWidth, halfWidth }) => {
```
Comment (bakycodixis) [inline components/TextInput.tsx]: If you have many props on a single line, break them onto separate lines for ease of readability.

File: app/page.tsx (hunk 5 above)
Comment (javiercodixis) [reply app/page.tsx]: everything is working fine, I just get this warning in the console when I start to fill the textinput fields

`Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. `

Comment (bakycodixis) [reply app/page.tsx]: You need to account for that. Your `field` object inside `TextInput` has a `value` property. Right now you spread the `field` onto `TextInput` which results in this component accepting whatever value is passed by `value` property. You need to make sure that if is present, then you pass it as is. Else (if non-existent, like null), then you pass an empty string.

**Review by bakycodixis — APPROVED** [review]

**Pull request merged**